core/trino-main/src/main/java/io/trino/server/security/oauth2/ScribeHttpClient.java

```diff
@@ -0,0 +1,197 @@
+            throws InterruptedException, ExecutionException, IOException
+    {
+        throw new UnsupportedOperationException("Not implemented");
+    }
+
+    @Override
+    public void close()
+            throws IOException
+    {
+        httpClient.close();
+    }
+
+    private Response execute(
+            String userAgent,
+            Map<String, String> headers,
+            Verb httpVerb,
+            String completeUrl,
+            BodyGenerator bodyGenerator)
+    {
+        ImmutableMultimap.Builder<String, String> requestHeaders = ImmutableMultimap.<String, String>builder()
+                .putAll(DEFAULT_HEADERS)
+                .putAll(headers.entrySet());
+        if (userAgent != null) {
+            requestHeaders.put(USER_AGENT, userAgent);
+        }
+        Request request = Request.builder()
+                .setMethod(httpVerb.name())
+                .setUri(URI.create(completeUrl))
+                .setBodyGenerator(bodyGenerator)
+                .addHeaders(requestHeaders.build())
+                .build();
+        return httpClient.execute(request, new ConvertingResponseHandler());
+    }
+
+    private static class ConvertingResponseHandler
+            implements ResponseHandler<Response, RuntimeException>
+    {
+        @Override
+        public Response handleException(Request request, Exception exception)
+        {
+            throw propagate(request, exception);
+        }
+
+        @Override
+        public Response handle(Request request, io.airlift.http.client.Response response)
+        {
+            return convertResponse(response);
+        }
+
+        private static Response convertResponse(io.airlift.http.client.Response response)
+        {
+            int code = response.getStatusCode();
+            Map<String, String> headers = response.getHeaders().asMap()
+                    .entrySet().stream()
+                    .collect(toImmutableMap(
+                            e -> e.getKey().toString(),
+                            e -> String.join(",", e.getValue())));
+            try {
+                return new Response(
+                        code,
+                        HttpStatus.fromStatusCode(code).toString(),
+                        headers,
+                        // the input stream must be read in the response handler
+                        getBody(response));
+            }
+            catch (IOException e) {
+                throw new UncheckedIOException(e);
+            }
+        }
+
+        private static String getBody(io.airlift.http.client.Response response)
+                throws IOException
+        {
+            // from com.github.scribejava.core.model.Response#parseBodyContents
+            InputStream stream = response.getInputStream();
+            if (stream == null) {
+                return null;
+            }
+            if ("gzip".equals(response.getHeader("Content-Encoding"))) {
```

Comment:
This is not needed, since content decoding would be done by the HTTP client. (though we won't ever get gzip anyway, since we configure the client not to request it)